src/Shared/Recruit.Vacancies.Client/Application/Rules/BaseRules/BaseBannedPhraseChecksRule.cs

```diff
@@ -0,0 +1,98 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Linq.Expressions;
+using System.Threading.Tasks;
+using Esfa.QA.Core.Extensions;
+using Esfa.Recruit.Vacancies.Client.Application.Providers;
+using Esfa.Recruit.Vacancies.Client.Application.Rules.Engine;
+using Esfa.Recruit.Vacancies.Client.Domain.Entities;
+
+namespace Esfa.Recruit.Vacancies.Client.Application.Rules.BaseRules
+{
+    public class BaseBannedPhraseChecksRule : Rule
+    {
+        private readonly IBannedPhrasesProvider _bannedPhrasesProvider;
+        private readonly BaseProfanityChecksRule.ConsolidationOption _consolidationOption;
+
+        public BaseBannedPhraseChecksRule(string ruleId, IBannedPhrasesProvider bannedPhrasesProvider, BaseProfanityChecksRule.ConsolidationOption consolidationOption, decimal weighting = 1.0m) 
+            : base(ruleId, weighting)
+        {
+            _bannedPhrasesProvider = bannedPhrasesProvider;
+            _consolidationOption = consolidationOption;
+        }
+
+        protected async Task<IEnumerable<RuleOutcome>> BannedPhraseCheckAsync(Expression<Func<string>> property, string relatedFieldId = null)
+        {
+            var fieldId = relatedFieldId ?? property.GetQualifiedFieldId();
+
+            var foundBannedPhrases = await FindOccurrencesAsync(property);
+
+            if (foundBannedPhrases.Values.Sum() > 0)
+                switch (_consolidationOption)
+                {
+                    case BaseProfanityChecksRule.ConsolidationOption.NoConsolidation:
+                        return CreateUnconsolidatedOutcomes(foundBannedPhrases, fieldId);
+
+                    case BaseProfanityChecksRule.ConsolidationOption.ConsolidateByField:
+                        return CreateConsolidatedOutcomes(foundBannedPhrases, fieldId);
+
+                    default:
+                        throw new ArgumentOutOfRangeException();
+                }
+
+            return new[] { CreateOutcome(0, $"No banned phrases found in '{fieldId}'", null, fieldId) };
+        }
+
+        private async Task<Dictionary<string, int>> FindOccurrencesAsync(Expression<Func<string>> property)
+        {
+            var value = property.Compile()();
+            var checkValue = value.FormatForParsing();
+
+            var foundBannedPhrases = new Dictionary<string, int>();
+            var phrases = await _bannedPhrasesProvider.GetBannedPhrasesAsync();
```

Comment:
Any way we can cache this in a private variable? I noticed with the Profanity check we're re-fetching the list for every property we are checking.  Appreciate it would be a bit tricky being async.